File: airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3DestinationConfigFactory.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.s3;

import com.fasterxml.jackson.databind.JsonNode;
import javax.annotation.Nonnull;

public class S3DestinationConfigFactory {

  public S3DestinationConfig getS3DestinationConfig(final JsonNode config, @Nonnull final StorageProvider storageProvider) {
    return S3DestinationConfig.getS3DestinationConfig(config, storageProvider);
  }

}

File: airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3DestinationConfigFactory.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.integrations.destination.s3

import com.fasterxml.jackson.databind.JsonNode
import javax.annotation.Nonnull

open class S3DestinationConfigFactory {
    open fun getS3DestinationConfig(
        config: JsonNode,
        @Nonnull storageProvider: StorageProvider,
        environment: Map<String, String>
    ): S3DestinationConfig {
        return S3DestinationConfig.Companion.getS3DestinationConfig(
            config = config,
            storageProvider = storageProvider,
            environment = environment
        )
    }
}
